docs/adr/adr_0012.md

```diff
@@ -0,0 +1,129 @@
+
+Potential workarounds here would be to introduce an additional field to the CascadingRule spec which allows us to define a sort of "deduplication" / replacement mechanism.
+
+### `ClusterScanCompletionHook`
+
+`ClusterScanCompletionHook` behave mostly like their namespaced counterpart, when the parser completes the operator fetches the list of ScanCompletionHooks from Scans namespace (respecting the hookSelector of the Scan) and the `ClusterScanCompletionHook`s (also respecting the hookSelector of the Scan (this might have to be discussed...)) are merged and ordered according to the respective priorities set on them.
+
+The execution of the hooks is different to the current model, as there be an additional setting on the `ClusterScanCompletionHook` called namespaceMode which allows users to configure to either run the hook in the same namespace as the scan it is getting executed for, or to provide a fixed namespace where all executions for this `ClusterScanCompletionHook` will be run in.
+
+Example `ClusterScanCompletionHook` specs with this being set:
+
+```yaml
+apiVersion: "execution.securecodebox.io/v1"
+kind: ClusterScanCompletionHook
+metadata:
+  name: slack-notification
+spec:
+  type: ReadAndWrite
+  image: "docker.io/securecodebox/notification"
+```
+
+```yaml
+apiVersion: "execution.securecodebox.io/v1"
+kind: ClusterScanCompletionHook
+metadata:
+  name: example-cluster-hook
+spec:
+  # spec.executionNamespace - would let you set a fixed namespace for the hook to get executed in
+  executionNamespace: example-namespace-name
+  type: ReadAndWrite
+  image: "docker.io/securecodebox/notification"
+```
+
+This fixed namespace mode has been added to the `ClusterScanCompletionHook` but not for the `ClusterScanType` for the following reasons:
+
+1. Hooks are usually used to interface with 3rd party systems which are often standardized for the entire company. Allowing to run them in a namespace separate from the individual teams / scans allows the hook access these systems without having to distribute and thus potentially compromise the credentials for the 3rd party systems.
+2. In contrast to scans, hooks usually do not need to send requests to the scanned applications. Scanning a app from outside their namespace opens up a lot of potential issues with network policies blocking inter network traffic between scanner and app. As the hooks only need to access the S3 bucket and the 3rd party system they are interfacing with, this is not a issue for hooks.
+
+#### Potential Issues With the Fixed `executionNamespace` Mode
+
+Can lead to inter team conflicts if a scans of "team 1" fails because of a error in the `ClusterScanCompletionHook` managed by a central cluster team. Members of "team 1" will likely not even have access to the namespace the failing hooks run in and wont be able to debug / fix these issues by their self.
```

Comment:
```suggestion
Can lead to conflicts between teams if a scan of "team 1" fails because of an error in the `ClusterScanCompletionHook` managed by a "central cluster team". Members of "team 1" will likely not even have access to the namespace the failing hooks run in and won't be able to debug / fix these issues by themselves.
```